src/scripts/main.js

```diff
@@ -1,3 +1,275 @@
+
+    return isNaN(a) && isNaN(b)
+      ? new Intl.Collator().compare(cellA, cellB) * direction
+      : (a - b) * direction;
+  });
+
+  tbody.append(...rowArray);
+});
+
+// selected row
+tbody.addEventListener('click', (e) => {
+  const rows = [...tbody.rows];
+
+  rows.forEach((row) => {
+    row.classList.remove('active');
+  });
+  e.target.parentElement.classList.add('active');
+});
+
+// form
+const body = document.querySelector('body');
+const form = document.createElement('form');
+
+body.append(form);
+form.classList.add('new-employee-form');
+
+// Name input
+const nameLabel = document.createElement('label');
+const nameInput = document.createElement('input');
+
+nameLabel.setAttribute('data-qa', 'name');
+nameLabel.innerText = 'Name:';
+form.append(nameLabel);
+
+nameInput.setAttribute('name', 'name');
+nameInput.setAttribute('type', 'text');
+nameLabel.append(nameInput);
+
+// position input
+const positionLabel = document.createElement('label');
+const positionInput = document.createElement('input');
+
+positionLabel.setAttribute('data-qa', 'position');
+positionLabel.innerText = 'Position:';
+form.append(positionLabel);
+
+positionInput.setAttribute('name', 'position');
+positionInput.setAttribute('type', 'text');
+positionLabel.append(positionInput);
+
+// office select
+const cities = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+const selectLabel = document.createElement('label');
+const selectOffice = document.createElement('select');
+
+selectLabel.setAttribute('data-qa', 'office');
+selectLabel.innerText = 'Office:';
+form.append(selectLabel);
+
+selectOffice.setAttribute('data-qa', 'office');
+selectOffice.setAttribute('name', 'office');
+selectLabel.append(selectOffice);
+
+cities.forEach((citi) => {
+  const optionCiti = document.createElement('option');
+
+  optionCiti.setAttribute('value', citi);
+  optionCiti.innerText = citi;
+  selectOffice.append(optionCiti);
+});
+
+// age input
+const ageLabel = document.createElement('label');
+const ageInput = document.createElement('input');
+
+ageLabel.setAttribute('data-qa', 'age');
+ageLabel.innerText = 'Age:';
+form.append(ageLabel);
+
+ageInput.setAttribute('name', 'age');
+ageInput.setAttribute('type', 'number');
+ageLabel.append(ageInput);
+
+// salary input
+const salaryLabel = document.createElement('label');
+const salaryInput = document.createElement('input');
+
+salaryLabel.setAttribute('data-qa', 'salary');
+salaryLabel.innerText = 'Salary:';
+form.append(salaryLabel);
+
+salaryInput.setAttribute('name', 'salary');
+salaryInput.setAttribute('type', 'number');
+salaryLabel.append(salaryInput);
+
+// button form
+const button = document.createElement('button');
+
+button.setAttribute('name', 'saveToTable');
+button.setAttribute('type', 'submit');
+button.innerText = 'Save to table';
+form.append(button);
+
+// button submit
+const formData = document.querySelector('.new-employee-form');
```

Comment:
The `formData` variable is assigned using `document.querySelector('.new-employee-form')`, but it should be `form` since the form is already created and appended to the body. This could lead to a runtime error if the form is not found in the DOM at the time of assignment.